docker/scripts/setup_env

```diff
@@ -44,34 +50,40 @@ DATABASE_ENCODING=${HUGINN_DATABASE_ENCODING:-${DATABASE_ENCODING}}
 DATABASE_PASSWORD=${HUGINN_DATABASE_PASSWORD:-${DATABASE_PASSWORD:-password}}
 DATABASE_NAME=${HUGINN_DATABASE_NAME:-${DATABASE_NAME:-huginn_production}}
 RAILS_ENV=${HUGINN_RAILS_ENV:-${RAILS_ENV:-production}}
+RAILS_LOG_TO_STDOUT=${RAILS_LOG_TO_STDOUT:-true}
+RAILS_SERVE_STATIC_FILES=${RAILS_SERVE_STATIC_FILES:-true}
+IP="${IP:-0.0.0.0}"
+PORT=${HUGINN_PORT:-${PORT:-3000}}
+PORT="$(echo $PORT|awk -F ':' '{print $NF}')"
+DOMAIN=${DOMAIN:-$IP:$PORT}
 
 # Add ENV variables to .env.example which are not present in it but usable
-cat >> /app/.env.example <<EOF
+cp -a /app/.env.example /tmp/
+cat >> /tmp/.env.example <<EOF
 ASSET_HOST=
 DEFAULT_SCENARIO_FILE=
+IP=
+RAILS_ENV=
+RAILS_LOG_TO_STDOUT=
 RAILS_SERVE_STATIC_FILES=
 SEED_EMAIL=
 SEED_PASSWORD=
 SEED_USERNAME=
 SMTP_OPENSSL_CA_FILE=
 SMTP_OPENSSL_CA_PATH=
 SMTP_OPENSSL_VERIFY_MODE=
+START_MYSQL=
 EOF
 
 # Default to the environment variable values set in .env.example
 IFS="="
-sed -n -r -e 's/^#?([A-Za-z0-9_]+ *=.*)/\1/p' /app/.env.example | \
+sed -n -r -e 's/^ *#?([A-Za-z0-9_]+=.*)/\1/p' /tmp/.env.example | \
   while read var value ; do
-    eval "echo \"$var=\${HUGINN_$var:-\${$var:-\$value}}\""
+    value="$(sed -e 's/^["'\'']//' -e 's/["'\'']$//' <<<"$value")"
```

Comment:
Can you explain what this line does?

Reply:
The right-hand side sometimes needs to be quoted. But then if `value` is from `.env.example` (vs. being passed in as environment variables), it may already be quoted. This line removes the leading/trailing quotes.

Reply:
Nice! I think that had that problem once or twice before 😄